src/cryptoadvance/specter/util/descriptor.py

```diff
@@ -171,11 +187,19 @@ def parse(cls, desc, testnet=False):
                     keys.sort(key=lambda x: x if "]" not in x else x.split("]")[1])
             multisig_M = desc.split(",")[0].split("(")[-1]
             multisig_N = len(keys)
+            if int(multisig_M) > multisig_N:
+                raise SpecterError(
+                    f"Multisig threshold cannot be larger than the number of keys. Threshold is {int(multisig_M)} but only {multisig_N} keys specified."
+                )
         else:
             keys = [desc.split("(")[-1].split(")", 1)[0]]
 
         descriptors = []
         for key in keys:
+            origin_fingerprint = None
```

Comment:
@stepansnigirev that fixed the "origin propagation" problem

Reply:
Awesome! Thanks!